doc/examples/example2.md

```diff
@@ -5,13 +5,16 @@ Certification Authority
 
 Chain of Trust 
 --------------
-
--Root Certificate-->Intermediate Certificate-->Client Certificate
+The example bellow shows how to create a self-signed root certificate with an RSA key
+and use it to sign a new certificate with a CSR
 
 
+Generate an RSA key pair
 \code{.cpp}
 AsymmetricKeypair rootRSAKey = AsymmetricKeypair::generateRSA();
-
+\endcode
+Set the desired values on the distinguished name for the self-signed certificate
```

Comment:
Don't break the code sample. Just add the explanation as comments inside the comments.

Reply:
Done.